pkg/controller/unifiedpushserver/util.go

```diff
@@ -63,19 +64,27 @@ func updateContainerSpecImage(deployment *appsv1.Deployment, name string, image
 	}
 }
 
-// apiVersionExists checks if a given API version exists in Kubernetes cluster.
-// Modified from https://github.com/operator-framework/operator-sdk/blob/947a464dbe968b8af147049e76e40f787ccb0847/pkg/k8sutil/k8sutil.go#L93
-// The Operator Framework one checks a specific resource exists, but this function checks if an API version exists.
-// Theoretically, there can be 2 resources in an API version, 1 exists and 1 not.
-func apiVersionExists(dc discovery.DiscoveryInterface, apiGroupVersion string) (bool, error) {
-	apiLists, err := dc.ServerResources()
-	if err != nil {
-		return false, err
-	}
-	for _, apiList := range apiLists {
-		if apiList.GroupVersion == apiGroupVersion {
-			return true, nil
+// apiVersionChecker is just a container to make it easier to fake the
+// check function for tests
+type apiVersionChecker struct {
```

Comment:
Maybe something with interfaces would be better here? I don't know

Reply:
Should it be in its own file if it's a struct with an interface?

Reply:
I've put it into its own file now :+1: